### PR TITLE
add configuration options to mod_security

### PR DESCRIPTION
**Pull request by jlambert121**
Add additional configuration options to mod_security
Ensure directories are purged to prevent duplicate definitions

### CODE REVIEW

**Comment by VolJin** [comment]
and please add:  notify  => Class['apache::service'],
to manage dirs:
$modsec_dir
"${modsec_dir}/activated_rules"

**Comment by jlambert121** [comment]
Yeah, that should be there.  Thanks for catching that!  Updated

File: manifests/mod/security.pp [hunk 1]
```diff
@@ -1,7 +1,11 @@
...
-  $crs_package     = $::apache::params::modsec_crs_package,
-  $activated_rules = $::apache::params::modsec_default_rules,
-  $modsec_dir      = $::apache::params::modsec_dir,
+  $crs_package         = $::apache::params::modsec_crs_package,
+  $activated_rules     = $::apache::params::modsec_default_rules,
+  $modsec_dir          = $::apache::params::modsec_dir,
+  $allowed_methods     = 'GET HEAD POST OPTIONS',
+  $content_types       = 'application/x-www-form-urlencoded|multipart/form-data|text/xml|application/xml|application/x-amf',
+  $restrict_extensions = '.asa/ .asax/ .ascx/ .axd/ .backup/ .bak/ .bat/ .cdx/ .cer/ .cfg/ .cmd/ .com/ .config/ .conf/ .cs/ .csproj/ .csr/ .dat/ .db/ .dbf/ .dll/ .dos/ .htr/ .htw/ .ida/ .idc/ .idq/ .inc/ .ini/ .key/ .licx/ .lnk/ .log/ .mdb/ .old/ .pass/ .pdb/ .pol/ .printer/ .pwd/ .resources/ .resx/ .sql/ .sys/ .vb/ .vbs/ .vbproj/ .vsdisco/ .webinfo/ .xsd/ .xsx/',
+  $restrict_headers    = '/Proxy-Connection/ /Lock-Token/ /Content-Range/ /Translate/ /via/ /if/',
```
Comment (igalic) [inline manifests/mod/security.pp]: uhm… what about… ✨documentation✨?

**Comment by jlambert121** [comment]
Documentation added

**Comment by igalic** [comment]
thanks @jlambert121!

**Pull request merged**